### PR TITLE
Improvement of CopyPasteManager.

### PR DESCRIPTION
**Pull request by rudyhuyn**
- Reject expressions like '4 8 15' (the app ignored spaces and read `4815`)
  - Except with the Programmer mode
- Accept 16e23 and not only 16e+23
- ignore more white spaces characters
- remove `Calculator::IsValidRegularExpression` (was buggy and limited and not used)
- Simplify the prototype of `ValidatePasteExpression` (return a bool instead of a string we don't need)
- simplify regex used by `CopyPasteManager`

### Changes
Instead of replacing all spaces, we replace all sequences of white spaces (tab, new line, etc...) by a single blank character `u+0020` so lexemes are not merged (`14 15` -> `1415`) and trim the start/end of the string.

`     14 +      16+12 ` -> `14 + 16+12`
### Questions

- Why do we accept `"` in digits?
- I suppose we still want to accept spaces with the Programmer mode (`100 111 110`), do we want to limit this behavior to all bases or limit it to `Binary`/`Hexa`/`Octo` bases ?

### How changes were validated:

Unit tests all passed + manual tests.

### CODE REVIEW

**Comment by danbelcher-MSFT** [comment]
Hi Rudy, thanks for the update.  I'm going to come back and give this a more thorough review but, in the meantime, can you please add unit tests for any new behavior you're adding here?

File: src/CalculatorUnitTests/CopyPasteManagerTest.cpp [hunk 1]
```diff
@@ -89,9 +89,7 @@ namespace CalculatorUnitTests
             vector<wstring> oneOperand = { L"123456" };
             VERIFY_ARE_EQUAL(m_CopyPasteManager.ExtractOperands(L"123456", ViewMode::Standard), oneOperand);
             oneOperand = { L"123^456" };
-            VERIFY_ARE_EQUAL(m_CopyPasteManager.ExtractOperands(L"123^456", ViewMode::Standard), oneOperand);
-
-            vector<wstring> twoOperands = { L"123", L"456" };
+            VERIFY_ARE_EQUAL(m_CopyPasteManager.ExtractOperands(L"123^456", ViewMode::Standard), oneOperand);            vector<wstring> twoOperands = { L"123", L"456" };
```
Comment (sanderl) [inline src/CalculatorUnitTests/CopyPasteManagerTest.cpp]: >vector<wstring> twoOperands = { L"123", L"456" }; [](start = 121, length = 49)

Can you put this on it's own line?

Comment (rudyhuyn) [reply src/CalculatorUnitTests/CopyPasteManagerTest.cpp]: I will as soon as I have access to a PC

**Comment by rudyhuyn** [comment]
@danbelcher-MSFT: I already modified the existing unit tests to cover the changes

**Comment by KillyMXI** [comment]
Space is used as a grouping separator in Russian locale. (Seldom in fact, but can't discard that.)

**Comment by HowardWolosky** [comment]
Thanks Rudy.  I'm looking forward to seeing your contributions get in.

This PR includes a _behavior change_ to the app, and our team is standardizing on a flow where a _behavior change_ needs to have an associated Issue which will provide the opportunity for discussion and approval before we'll consider merging in the corresponding code change.

We are in the process of updating CONTRIBUTING.md and our PR templates to clarify this.  In the meantime, please open an Issue to track this so that we can start the discussion.  Our team members will be able to join in on the discussion on Monday.

**Comment by HowardWolosky** [comment]
Locked this PR for further dicussion until it is associated with an approved Issue.

**Pull request closed without merging**